src/main/java/com/spotify/netty4/handler/codec/zmtp/ZMTPFramingEncoder.java

```diff
@@ -91,48 +89,22 @@ public void flush(final ChannelHandlerContext ctx) throws Exception {
       encoder.encode(message, writer);
       ReferenceCountUtil.release(message);
     }
-    final ChannelPromise aggregate = new AggregatePromise(ctx.channel(), promises);
+    final List<ChannelPromise> sentPromises = new ArrayList<ChannelPromise>(promises);
     messages.clear();
     promises.clear();
-    ctx.write(output, aggregate);
-    ctx.flush();
-  }
-
-  private static class AggregatePromise extends DefaultChannelPromise {
```

Comment:
The rationale for `AggregatePromise` was to save some garbage. Do we need to do away with it?